test/export/InstanceExporter.test.ts

```diff
@@ -1330,6 +1330,75 @@ describe('InstanceExporter', () => {
       ]);
     });
 
+    it('should assign fixed value[x] to the correct path when the rule on the instance refers to value[x], and value[x] is constrained to one type', () => {
+      // Profile: ObservationProfile
+      // Parent: Observation
+      // * value[x] 1..1
+      // * value[x] only string
+      const observationProfile = new Profile('ObservationProfile');
+      observationProfile.parent = 'Observation';
+      const valueCard = new CardRule('value[x]');
+      valueCard.min = 1;
+      valueCard.max = '1';
+      const valueOnly = new OnlyRule('value[x]');
+      valueOnly.types = [{ type: 'string' }];
+      observationProfile.rules.push(valueCard, valueOnly);
+      doc.profiles.set(observationProfile.name, observationProfile);
+      // Instance: MyObservation
+      // InstanceOf: ObservationProfile
+      // * status = #final
+      // * code = #testcode
+      // * value[x] = "some value"
+      const testInstance = new Instance('MyObservation');
+      testInstance.instanceOf = 'ObservationProfile';
+      const statusFinal = new AssignmentRule('status');
+      statusFinal.value = new FshCode('final');
+      const codeTestCode = new AssignmentRule('code');
+      codeTestCode.value = new FshCode('testcode');
+      const valueSomeValue = new AssignmentRule('value[x]');
+      valueSomeValue.value = 'some value';
+      testInstance.rules.push(statusFinal, codeTestCode, valueSomeValue);
+      const exported = exportInstance(testInstance);
+      expect(exported.valueString).toBe('some value');
+      expect(exported['value[x]']).toBeUndefined();
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
+      expect(loggerSpy.getLastMessage('warn')).toMatch(
+        "When assigning values on instances, use the choice element's type. Rule path changed to valueString"
```

Comment:
In the context of this PR I know exactly what you're talking about.  But I wonder if authors might be confused (until they look up the line numbers).  I think it might be more obvious what is going on if we also refer to the old path.  E.g.:
> When assigning values on instances, use the choice element's type. Rule path changed from 'value[x]' to 'valueString'.